muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/profile.js

```diff
@@ -1,7 +1,53 @@
 /* global moment */
+/* global MUIKKU_LOGGED_USER */
 
 (function() {
   
+  $.widget("custom.chatVisibility", {
+    _create: function() {
+      mApi().chat.status.read({}).callback($.proxy(function (err, status) {
+        if (err) { 
+          $('.notification-queue').notificationQueue('notification', 'error', err);
+          return;
+        }
+        
+        if (!status.chatSettingsVisible) { 
+          return; 
+        }
+        
+        mApi().chat.settings.read({}).callback($.proxy(function (err, settings) {
+          var data = {};
+          if (settings == null || settings.visibility == null){
+            settings.visibility === "DISABLED"
+            data.disabled_selected = "selected";
+            this._setVisibility("DISABLED");
+          }
+          if (settings && settings.visibility === "VISIBLE_TO_ALL") {
+            data.visible_to_all_selected = "selected";
+          }
+          if (settings && settings.visibility === "DISABLED"){
+            data.disabled_selected = "selected";
+          }
+          renderDustTemplate('profile/profile-chat-visibility.dust', data, $.proxy(function (text) {
+            this.element.html(text);
+            this.element.find("select").on('change', $.proxy(function(event) {
+              this._setVisibility(event.target.value);
+              setTimeout(function(){ location.reload(); }, 1500);
+              $('.notification-queue').notificationQueue('notification', 'success', getLocaleText("plugin.profile.chat.visibilityChange"));
+            }, this));
+          }, this));
+        }, this));
+      }, this));
+    },
+    
+    _setVisibility: function(visibility) {
+      var userIdentifier = MUIKKU_LOGGED_USER;
+      mApi().chat.settings.update({visibility: visibility, userIdentifier: userIdentifier}).callback($.proxy(function () {
+     
```

Comment:
...and simply do `location.reload();` in this callback method as it is called as soon as the server-side has stored the new value.